src/kOS.Safe/Execution/CPU.cs

```diff
@@ -413,11 +415,13 @@ public void AssertValidDelegateCall(IUserDelegate userDelegate)
         {
             if (userDelegate.ProgContext != currentContext)
             {
-                throw new KOSInvalidDelegateContextException(
-                    (currentContext == contexts[0] ? "the interpreter" : "a program"),
-                    (currentContext == contexts[0] ? "a program" : "the interpreter")
-                    );
-            }
+                string currentContextName = (currentContext == contexts[0] ? "the interpreter" : "a program");
+                string delegateContextName = (
+                    userDelegate.ProgContext == contexts[0] ? currentContextName = "the interpreter" : (
+                        currentContext == contexts[0] ? "a program" : "a different program from a previous run" ) );
+
+                throw new KOSInvalidDelegateContextException(currentContextName, delegateContextName);
+           }
```

Comment:
My issue is a matter of readability, multiple chains like this just take more attention to review.  Here is how I interpret what you have submitted:
```
                string delegateContextName;
                if (userDelegate.ProgContext == contexts[0])
                {
                    currentContextName = "the interpreter";
                    delegateContextName = "the interpreter";
                }
                else if (currentContext == contexts[0])
                {
                    delegateContextName = "a program";
                }
                else
                {
                    delegateContextName = "a different program from a previous run";
                }
```